packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -41,7 +41,7 @@ export class ApplicationExtension<
 
         if (!isServerSide()) {
             // Deserialize the routes on the client to ensure the latest routes are loaded on the client
-            this._cachedRoutes = this.deserialize()?.routes || null
+            this._cachedRoutes = window.__EXTENSIONS__[this.getName()]?.routes || null
```

Comment:
this means, extensions with a `getRoutesAsync` implementation will always have to have a `beforeRouteMatch implementation that deserializes the routes themselves on the client.

Reply:
Hmm right 🤔 I think I see what you mean there..

My thinking of deserialize is too simplistic. Not only we need to get the routes saved in `window`, but also we need to reconstruct the route's component.